### DISCUSSION TITLE
OpenHantek for 6022BE on Windows XP

### DISCUSSION THREAD

**Discussion by DVEHUB72** [OpenHantek/OpenHantek6022#174] [Q&A] [answered]
I am attempting to install OpenHantek on a Lenovo T61 laptop with Windows XP, (32 bit). I have tried two versions, OpenHantek-Win-x86-Release-b488.zip and OpenHantek-Win-x86-Release-b341.zip. Both result in the following message:
"not a valid Win32 application".  Any advice on how to solve this problem? Thanks in advance.

**Comment by Ho-Ro** [comment]
Older releases < 3.2 used Visual Studio 2015 on [appveyor](https://ci.appveyor.com/project/Ho-Ro/openhantek6022/history) and provided x86 and x64 builds. Starting with 3.2 I changed to Visual Studio 2019 and dropped the x86 binary (due to build issues with VS2019). 
Microsoft stopped the support for XP in 2014 and I suspect that your system is too old even for VS2015 builds. :(

Three solutions for your issue come to my mind:
1. Either you upgrade your system to at least Win7 (that was reported to be much more stable than XP or Vista).
2. Or a MS guru here provides a backward compatible solution without any regressions for up-to-date systems.
3. Or you do the big step and install Linux - but you have to build OH for yourself as my packages are only 64 bit. I did this test wise on an old PentiumM laptop a year ago without big issues.

**Reply by DVEHUB72** [reply]
I tried solution #1 by using another laptop with Windows 10. The installation went the same as on my desktop but when I connected the scope and tried to run it I received the message: "This app can't run on your PC". No clue why not.
Solution #2 would be nice if someone comes forward; I'll keep monitoring this discussion.
Solution #3 My past experience with Linux was not positive. This would be a last resort.

**Reply by DVEHUB72** [reply]
I have done a clean install of Windows 10 64 bit on the laptop. I then installed OpenHantek on it. Now I do not receive the message that the app won't run on it. However, when I start the program the select screen is displayed for about a half second and then closes. If I disconnect the DSO the select screen is displayed but when I click on the demo mode it closes. I checked Device Manager and verified that the driver files are the same as on my desktop, (where OpenHantek runs fine). What should I try next?

**Reply by Ho-Ro** [reply]
You can call OpenHantek from a cmd.exe window to see if it emits some error messages. 
This message is not critical:
````
QObject::connect: Cannot queue arguments of type 'QList<double>'
(Make sure 'QList<double>' is registered using qRegisterMetaType().)
````
Did you install the supplied `vcredist_x64.exe`? [support.microsoft.com: the-latest-supported-visual-c-downloads](https://support.microsoft.com/en-us/topic/the-latest-supported-visual-c-downloads-2647da03-1eea-4433-9aff-95f26a218cc0)

**Reply by DVEHUB72** [reply]
I did install vcredist_x64.exe. When I used the command prompt to call OpenHantek I received the following response:

QwindowsEGLContext: ANGLE only partially supports OpenGL ES > 3.0
QwindowsEGLContext: Failed to create context, eglError: 3005, this: 0x186a4d56460
QOpenGLFunctions created with non-current context

I don't know what this means or what to do about it. Advice?

**Reply by Ho-Ro** [reply]
The error comes from OpenGL graphics - you could try these command line options (one at a time):
````
  -e, --useGLES                Use OpenGL ES instead of OpenGL
  --useGLSL120                 Force OpenGL SL version 1.20
  --useGLSL150                 Force OpenGL SL version 1.50
````
If this doesn't work you should check the OpenGL version on your system. What I just did on a W10 system is to check the properties of C:\Windows\System32\opengl32.dll and it said: `Version 10.0.18362.1500` but this is only the dll version, it doesn't tell which OpenGL version is supported on my system (depending on the HW).
A quick search found [this tool](https://www.realtech-vr.com/home/glview).

EDIT:
You can also check without external tool: `set QT_LOGGING_RULES=qt.qpa.gl=true` and call Openhantek from command window. This will give a lot of output but should give also more hints why the program fails.

**Comment by Ho-Ro** [comment, marked as answer]
See also #184

**Reply by DVEHUB72** [reply]
Thanks for your advice. I tried the command `OpenHantek.exe –useGLES` (EDIT Ho-Ro: formatting) and this was displayed:

QObject::connect: Cannot queue arguments of type 'QList<double>'
(Make sure 'QList<double>' is registered using qRegisterMetaType().)

Despite this, the main screen for Openhantek opened with the correct channel 1 trace. (The test signal).

The reference to QObject and QList looks like names from QtCreator. Is this right? I noticed that QtCreator was recommended for building. I'm not sure I want to go this far but if so, where would I find the source files?

**Reply by Ho-Ro** [reply]
> The reference to QObject and QList looks like names from QtCreator. Is this right?

No, [QObject](https://doc.qt.io/qt-5/qobject.html) and [QList](https://doc.qt.io/qt-5/qlist.html) are classes of Qt, the cross-platform software development framework, while QtCreator is the IDE (and CMake is the make system). The use of Qt allows to develop software, that builds and runs not only on Linux (where I develop) but also on FreeBSD, macOS and even on Windows. 

But as I wrote above: 
> This message is not critical:
````
QObject::connect: Cannot queue arguments of type 'QList<double>'
(Make sure 'QList<double>' is registered using qRegisterMetaType().)
````
> where would I find the source files?

Just [around the corner](https://github.com/OpenHantek/OpenHantek6022/tree/main/openhantek/src). If you want to deep dive, either clone the repo if you want to use git or download the zipped source.
![image](https://user-images.githubusercontent.com/12542359/115967271-fce86000-a531-11eb-8a45-fe0a53eda8aa.png)

**Comment by MaMiKoe** [comment]
I  have made the same experience when tyring to use the software with WinXP SP3.  Although I really do not know anything about VS (and I am also not able to program in C) I doubt that the problem is WinXP itself - as far as I understand (and as said I do not know anyhting about VS) VS 2015 shoud be compatible with WinXP: https://docs.microsoft.com/en-us/visualstudio/releases/2015/vs2015-compatibility-vs.
An other question is whether an old and slow WinXP-PC ist fast enough for the software.
But my experiences say that an "old" WinXP-PC runs with WinXP nearly so fast as a newer Win7-PC with Win7 and an more newer Win10-PC with Win10. 
When starting OpenHantek.exe (from OpenHantek-Win-x86-Release-b488.zip) in DOS/command windows  the (system-?) message is: "access denied". Maybe the reason is not VS 2015 itself (in combination with WinXP) but that OpenHantek.exe is not starting as a 32bit-application? When trying to run the 64Bit-version on WinXP-32Bit the same messages appear (not a valid 32Bit-application).
Of course installing the vcredist_x64.exe id not possible in an 32Bit-environment. But the not-a-valid-32Bit-applications also results when vcredist_x86.exe is not installed, what leads to the assumption, that not the VS itself with WinXP is the problem.
